packages/theme-check-common/src/checks/valid-visible-if/visible-if-utils.ts

```diff
@@ -0,0 +1,139 @@
+import { type LiquidVariableLookup, toLiquidHtmlAST, NodeTypes } from '@shopify/liquid-html-parser';
+import type { Context, SourceCodeType } from '../..';
+import { findRoot } from '../../find-root';
+import { parseJSON } from '../../json';
+import { join } from '../../path';
+import { visit } from '../../visitor';
+
+export type Vars = { [key: string]: Vars | true };
+
+export const variableExpressionMatcher = /{{(.+?)}}/;
+export const adjustedPrefix = '{% if ';
+export const adjustedSuffix = ' %}{% endif %}';
+export const offsetAdjust = '{{'.length - adjustedPrefix.length;
+
+export async function getGlobalSettings(context: Context<SourceCodeType>) {
+  const globalSettings: string[] = [];
+
+  try {
+    const path = join(
+      await findRoot(context.file.uri, context.fileExists),
+      'config/settings_schema.json',
+    );
+    const settingsFile = await context.fs.readFile(path);
+    const settings = parseJSON(settingsFile);
+    if (Array.isArray(settings)) {
+      for (const group of settings) {
+        if ('settings' in group && Array.isArray(group.settings)) {
+          globalSettings.push(
+            ...group.settings.map((setting: any) => setting.id).filter((id: any) => id),
+          );
+        }
+      }
+    }
+  } catch (e) {
+    console.error('Error fetching global settings:', e);
+    // ignore absent or malformed settings schema
+  }
+
+  return globalSettings;
+}
+
+export function getVariableLookupsInExpression(
+  expression: string,
+): LiquidVariableLookup[] | { warning: string } {
+  // parsers other than liquidjs don't yet support expressions in {{ variable }}
```

Comment:
We have a few references in here to hacky/gnarly approach but it doesn't look like we explicitly say what that approach is. Considering what we're doing to make this work I think we should have a pretty robust comment explaining the approach because the next person to poke at this without context won't have much of an idea of what's happening.

Reply:
perfect!